File: tests/lang-test/javascript/src/models/product.js
// ripex-lang-test: JS product model — enum-ish, methods, computed names.
export const Category = {
  Electronics: 'electronics',
  Clothing: 'clothing',
};

export class Product {
  /**
   * @param {number} id
   * @param {string} name
   * @param {number} price
   * @param {string} category
   */
  constructor(id, name, price, category) {
    this.id = id;
    this.name = name;
    this.price = price;
    this.category = category;
  }

  get ['displayName']() {
    return `${this.name} (${this.category})`;
  }

  /** @param {number} rate */
  calculateTax(rate = 0.1) {
    return this.price * rate;
  }
}

File: tests/lang-test/python/src/models/product.py
"""ripex-lang-test: Python Product model — enum, class var, method."""
from enum import Enum
from typing import Optional


class Category(Enum):
    ELECTRONICS = "electronics"
    CLOTHING = "clothing"


class Product:
    tax_rate: float = 0.1

    def __init__(self, id: int, name: str, price: float, category: Category) -> None:
        self.id = id
        self.name = name
        self.price = price
        self.category = category

    def calculate_tax(self, rate: Optional[float] = None) -> float:
        return self.price * (rate if rate is not None else self.tax_rate)
